### PR TITLE
fix(rewrite): avoid wildcards on destination topic on publish

### PR DESCRIPTION
**Pull request by paulozulato**
Fixes https://emqx.atlassian.net/browse/EMQX-9247

### <samp>🤖 Generated by Copilot at 3a428ae</samp>

This pull request improves the validation and testing of the topic rewrite rules feature in the `emqx_modules` app. It adds a validator function to check the destination topics in the rules and a test case to verify the API response for invalid rules.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [x] Changed lines covered in coverage report
- [x] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] ~If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up~
- [x] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] ~If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)~
- [ ] ~Change log has been added to `changes/` dir for user-facing artifacts update~

### CODE REVIEW

File: changes/ce/fix-11004.en.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Avoid wildcards on destination topic for rewrite rules.
```
Comment (zmstone) [inline changes/ce/fix-11004.en.md:1]:
```suggestion
Do not allow wildcards for destination topic in rewrite rules.
```

File: apps/emqx_modules/src/emqx_modules_schema.erl [hunk 2]
```diff
@@ -45,6 +46,28 @@ roots() ->
...
+            fun
+                (#{<<"action">> := subscribe}, Acc) ->
+                    Acc;
+                (#{<<"dest_topic">> := DestTopic} = Rule, InvalidAcc) ->
+                    case catch emqx_topic:validate(name, string:trim(DestTopic)) of
+                        true ->
+                            InvalidAcc;
+                        _ ->
+                            [Rule | InvalidAcc]
+                    end
```
Comment (zmstone) [inline apps/emqx_modules/src/emqx_modules_schema.erl:61]: Please consider direct `catch` deprecated and we should never use it for new code in EMQX codebase.
```suggestion
                    try
                        true = emqx_topic:validate(name, DestTopic),
                        InvalidAcc
                    catch _:_ ->
                        [DestTopic | InvalidAcc]
                    end
```

File: apps/emqx_modules/src/emqx_modules_schema.erl [hunk 3]
```diff
@@ -45,6 +46,28 @@ roots() ->
...
+            end,
+            [],
+            Rules
+        )
+    of
+        [] -> ok;
+        InvalidRules -> {error, #{msg => "Invalid rules", invalid_rules => InvalidRules}}
```
Comment (zmstone) [inline apps/emqx_modules/src/emqx_modules_schema.erl:68]:
```suggestion
        InvalidRules -> {error, #{msg => "cannot_use_wildcard_for_destination_topic", invalid_topics => InvalidTopics}}
```

File: changes/ce/fix-11004.en.md (hunk 1 above)
Comment (paulozulato) [reply changes/ce/fix-11004.en.md]: Ack

File: apps/emqx_modules/src/emqx_modules_schema.erl (hunk 2 above)
Comment (paulozulato) [reply apps/emqx_modules/src/emqx_modules_schema.erl]: Ack

File: apps/emqx_modules/src/emqx_modules_schema.erl (hunk 3 above)
Comment (paulozulato) [reply apps/emqx_modules/src/emqx_modules_schema.erl]: Ack

**Review by zhongwencool — APPROVED** [review]

**Review by thalesmg — APPROVED** [review]

**Pull request merged**